storage: TestReplicateQueueRebalance failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/670c7d3df74e0efbc633ceac8e97fada3b1063b2

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestReplicateQueueRebalance PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=971695&tab=buildLog

```
W181017 07:56:36.101909 112812 storage/raft_transport.go:583  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W181017 07:56:36.102307 112337 storage/raft_transport.go:583  [n3] while processing outgoing Raft queue to node 4: rpc error: code = Canceled desc = grpc: the client connection is closing:
I181017 07:56:36.109222 114511 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181017 07:56:36.109448 114509 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
W181017 07:56:36.090991 112421 storage/raft_transport.go:583  [n4] while processing outgoing Raft queue to node 5: rpc error: code = Canceled desc = grpc: the client connection is closing:
I181017 07:56:36.110474 114510 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-subscription
I181017 07:56:36.110867 114508 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] transport racer
I181017 07:56:36.121011 114508 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
I181017 07:56:36.148469 114580 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n4: context canceled
I181017 07:56:36.149442 114550 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n4: context canceled
I181017 07:56:36.149918 114450 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n2: context canceled
I181017 07:56:36.151412 114549 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n3: context canceled
I181017 07:56:36.151667 114579 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n1: context canceled
I181017 07:56:36.152628 114449 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n5: context canceled
I181017 07:56:36.154413 114551 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n5: context canceled
I181017 07:56:36.158025 114548 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n1: context canceled
I181017 07:56:36.186170 114529 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n3: context canceled
I181017 07:56:36.186503 114611 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n1: context canceled
I181017 07:56:36.187847 114612 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n4: context canceled
I181017 07:56:36.188046 114530 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n2: context canceled
I181017 07:56:36.211143 114538 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n5: context canceled
I181017 07:56:36.213022 114535 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n3: context canceled
I181017 07:56:36.214498 114536 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n1: context canceled
I181017 07:56:36.216249 114537 rpc/nodedialer/nodedialer.go:91  [ct-client] unable to connect to n2: context canceled
W181017 07:56:36.252561 110210 gossip/gossip.go:1499  [n1] no incoming or outgoing connections
I181017 07:56:36.895936 110170 kv/transport_race.go:91  transport race promotion: ran 68 iterations on up to 2753 requests
--- FAIL: TestReplicateQueueRebalance (69.43s)
	soon.go:49: condition failed to evaluate within 45s: not balanced: [15 16 16 15 13]
		goroutine 109997 [running]:
		runtime/debug.Stack(0xa7a358200, 0xc4208e3110, 0x3bc3f20)
			/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
		github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3c20a40, 0xc42208e0f0, 0xc421ab6640)
			/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:50 +0x172
		github.com/cockroachdb/cockroach/pkg/storage_test.TestReplicateQueueRebalance(0xc42208e0f0)
			/go/src/github.com/cockroachdb/cockroach/pkg/storage/replicate_queue_test.go:100 +0x730
		testing.tRunner(0xc42208e0f0, 0x34bcf40)
			/usr/local/go/src/testing/testing.go:777 +0x16e
		created by testing.(*T).Run
			/usr/local/go/src/testing/testing.go:824 +0x565

```

--- a/pkg/storage/replicate_queue_test.go
+++ b/pkg/storage/replicate_queue_test.go
@@ -51,6 +51,8 @@ func TestReplicateQueueRebalance(t *testing.T) {
 		base.TestClusterArgs{
 			ReplicationMode: base.ReplicationAuto,
 			ServerArgs: base.TestServerArgs{
+				ScanMinIdleTime: time.Millisecond,
+				ScanMaxIdleTime: time.Millisecond,
 				Knobs: base.TestingKnobs{
 					Store: &storagebase.StoreTestingKnobs{
 						// Prevent the merge queue from immediately discarding our splits.
@@ -104,7 +106,8 @@ func TestReplicateQueueRebalance(t *testing.T) {
 		counts := countReplicas()
 		for _, c := range counts {
 			if c < minReplicas {
-				err := errors.Errorf("not balanced: %d", counts)
+				err := errors.Errorf(
+					"not balanced (want at least %d replicas on all stores): %d", minReplicas, counts)
 				log.Info(context.Background(), err)
 				return err
 			}

